src/components/molecules/SecretPasswordForm/SecretPasswordForm.tsx

```diff
@@ -1,34 +1,54 @@
 import React, { useCallback, useState } from "react";
-import { useHistory } from "react-router-dom";
 
 import { checkAccess } from "api/auth";
 
-import { venueEntranceUrl } from "utils/url";
 import { setLocalStorageToken } from "utils/localStorage";
 import { isTruthy } from "utils/types";
 
 import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
 
 import "./SecretPasswordForm.scss";
 
-const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
-  const history = useHistory();
+interface SecretPasswordFormProps {
+  buttonText: string;
+  onPasswordSubmit?: () => void;
+  onPasswordSuccess?: () => void;
+}
+
+export const SecretPasswordForm: React.FC<SecretPasswordFormProps> = ({
+  buttonText = "Join the party",
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const { user } = useUser();
   const venueId = useVenueId();
 
-  const [error, setError] = useState();
-  const [password, setPassword] = useState();
-  const [message, setMessage] = useState();
+  const [error, setError] = useState(false);
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
 
-  function passwordChanged(e) {
-    setPassword(e.target.value);
-    setMessage("");
-    setError(false);
-  }
+  const changePassword = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setPassword(e.target.value);
+      setMessage("");
+      setError(false);
+    },
+    []
+  );
```

Comment:
looks and sounds interesting but isn't this an anti pattern? Directly manipulating the DOM is not one of the best practices and should be used only if necessary.

Reply:
We're not directly manipulating the DOM, we are just getting a reference to the input element, and reading a value from it.

If you wanted to go a step further with it, perhaps we should be aligning all of our textbox/form/etc code to use our proper 'rect form handling' libraries, rather than 'doing it directly' with state/refs/etc. Which looking at our `package.json` is:

- https://react-hook-form.com/